src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -56,6 +56,21 @@ private static String str(double n) {
 		Parameter<?>[] numbersParam = new Parameter[] {new Parameter<>("ns", DefaultClasses.NUMBER, false, null)};
 		
 		// basic math functions
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("average", new Parameter[] {new Parameter<>("list", DefaultClasses.NUMBER, false, null)}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number[] executeSimple(Object[][] params) {
+				Double sum = 0.0;
+				int length = params[0].length;
+				for (int i = 0; i < length; i++) {
+					if (params[0][i] instanceof Number)
```

Comment:
No need to check if it's an instance of `Number`

Reply:
Added by Pickle :D

Reply:
It's an appropriate change. We talked over it. Skript already ensures the Object is of the appropriate type and that it's not null.

Reply:
@cooffeeRequired unresolved